### PR TITLE
Only show replication factor input when relevant

### LINKED ISSUE #64: Don't show replication factor input when offsets.topic.replication.factor=1
From https://github.com/jlandersen/vscode-kafka/pull/60#discussion_r551871879

Apparently, you can't use a replication factor higher than the cluster's `offsets.topic.replication.factor` value.

IMHO if `offsets.topic.replication.factor`=1, then there's no point in even showing that part of the wizard.

See https://github.com/jlandersen/vscode-kafka/blob/824a615fd5d49e2021b485d694dc9706429bc1a6/src/commands/topics.ts#L145-L156 as an example to get the cluster's config value:

@jlandersen what do you think about that?

### PR DESCRIPTION
**Pull request by fbricon**
When creating a new topic from the wizard

- read offsets.topic.replication.factor from broker config to determine default replication factor
- fallback to default.replication.factor
- fallback to the lowest value between 3 and the number of brokers (looking at you cloudkarafka)

The max number of replicas is determined by the number of brokers.

If value <= 1, the replication factor input is not shown in the topic wizard
If value > max replicas, a validation error is displayed

Fixes #64

### CODE REVIEW

File: src/wizards/topics.ts [hunk 1]
```diff
@@ -160,12 +175,44 @@ async function inputPartitions(input: MultiStepInput, state: Partial<CreateTopic
...
         title: INPUT_TITLE,
         step: input.getStepNumber(),
         totalSteps: state.totalSteps,
-        value: state.replicationFactor || DEFAULT_REPLICATION_FACTOR.toString(),
+        value: state.replicationFactor || state.maxReplicas!.toString(),
         prompt: 'Replication Factor',
         validate: validateReplicationFactor
     });
 }
+async function getMaxReplicas(clientAccessor: ClientAccessor, clusterId: string): Promise<number> {
```
Comment (fbricon) [inline src/wizards/topics.ts:191]: actually we need to distinguish default value and max value

File: src/wizards/topics.ts [hunk 2]
```diff
@@ -7,13 +7,17 @@ import { MultiStepInput, showErrorMessage, State } from "./multiStepInput";
 import { validateTopicName, validatePartitions, validateReplicationFactor } from "./validators";
 
 const DEFAULT_PARTITIONS = 1;
-const DEFAULT_REPLICATION_FACTOR = 1;
+
+const REPLICATION_FACTOR_KEY = 'offsets.topic.replication.factor';
```
Comment (angelozerr) [inline src/wizards/topics.ts:11]: Those constants are about broker configs (https://kafka.apache.org/documentation/#brokerconfigs). I will create a `src/client/configs.ts` file and define those constants with namespace (liek https://github.com/redhat-developer/vscode-xml/blob/27983849c3d19b7a53e29e66a4f128c1e09275d5/src/commands/commandConstants.ts#L16) :

```ts
/**
 * @see https://kafka.apache.org/documentation/#brokerconfigs
 */
export namespace BrokerConfigs {

    /**
    * @see https://kafka.apache.org/documentation/#brokerconfigs_default.replication.factor
    */
    export const DEFAULT_REPLICATION_FACTOR = 'default.replication.factor';

    /**
     * @see https://kafka.apache.org/documentation/#brokerconfigs_offsets.topic.replication.factor
     */
    export const OFFSETS_TOPIC_REPLICATION_FACTOR = 'offsets.topic.replication.factor';

}

```

Comment (fbricon) [reply src/wizards/topics.ts]: done

**Comment by angelozerr** [comment]
Thanks @fbricon !

**Pull request merged**